sklearn/metrics/tests/test_classification.py

```diff
@@ -2905,3 +2905,241 @@ def test_brier_score_loss_deprecation_warning():
+    y_pred = [[0.5, 0.5], [0.5, 0.5], [0.5, 0.5]]
+    labels = [0, 1, 2]
+    err = "number of classes in labels is different"
+    with pytest.raises(ValueError, match=err):
+        d2_log_loss_score(y_true, y_pred, labels=labels)
+
+    # check error if y_true and y_pred do not have equal lengths
+    y_true = [0, 1, 2]
+    y_pred = [[0.5, 0.5, 0.5], [0.6, 0.3, 0.1]]
+    err = "inconsistent numbers of samples"
+    with pytest.raises(ValueError, match=err):
+        d2_log_loss_score(y_true, y_pred)
+
+    # check warning for samples < 2
+    y_true = [1]
+    y_pred = [[0.5, 0.5]]
+    err = "score is not well-defined"
+    with pytest.warns(UndefinedMetricWarning, match=err):
+        d2_log_loss_score(y_true, y_pred)
+
+    # check error when y_true only has 1 label
+    y_true = [1, 1, 1]
+    y_pred = [[0.5, 0.5], [0.5, 0.5], [0.5, 5]]
+    err = "y_true contains only one label"
+    with pytest.raises(ValueError, match=err):
+        d2_log_loss_score(y_true, y_pred)
+
+    # check error when y_true only has 1 label and labels also has
+    # only 1 label
+    y_true = [1, 1, 1]
+    labels = [1]
+    y_pred = [[0.5, 0.5], [0.5, 0.5], [0.5, 5]]
+    err = "The labels array needs to contain at least two"
+    with pytest.raises(ValueError, match=err):
+        d2_log_loss_score(y_true, y_pred, labels=labels)
+
+    # Some tests with sample weights
+
+    # check if poor predictions for character values gives a relatively
+    # low value for the d2 score when sample weights are also given.
+    y_true = ["h", "d", "c", "c"]
+    sample_weight = [1.0, 0.5, 0.5, 1.0]
+    y_pred = np.array(
+        [
+            [0.6, 0.3, 0.1],
+            [0.5, 0.2, 0.3],
+            [0.3, 0.4, 0.3],
+            [0.4, 0.5, 0.1],
+        ]
+    )
+    d2_score = d2_log_loss_score(y_true, y_pred, sample_weight=sample_weight)
+    assert d2_score < 0
+
+    # check if good predictions for character values gives a relatively
+    # higher d2 score when sample weights are also given.
+    y_true = ["orange", "strawberry", "apple", "apple"]
+    sample_weight = [3, 2, 1, 4]
+    y_pred = np.array(
+        [
+            [0.1, 0.8, 0.1],
+            [0.1, 0.1, 0.8],
+            [0.8, 0.1, 0.1],
+            [0.8, 0.1, 0.1],
+        ]
+    )
+    d2_score = d2_log_loss_score(y_true, y_pred, sample_weight=sample_weight)
+    assert 0.5 < d2_score < 1.0
+
+    # check if simply using the average of the classes as the predictions
+    # gives a d2 score of 0 where the y_true values are characters and "okay"
+    # has a relatively higher proportion, when sample weights are also given.
+    y_true = ["low", "neutral", "high", "high"]
+    sample_weight = [2, 2, 2, 2]
+    y_pred = np.array(
+        [
+            [0.5, 0.25, 0.25],
+            [0.5, 0.25, 0.25],
+            [0.5, 0.25, 0.25],
+            [0.5, 0.25, 0.25],
+        ]
+    )
+    d2_score = d2_log_loss_score(y_true, y_pred, sample_weight=sample_weight)
+    assert d2_score == 0
```

Comment:
Could you remove it here and add the sample weights below the corresponding test above?

Reply:
Okay I'll try to make the tests a bit more concise.